src/language/dafnyInstallation.ts

```diff
@@ -153,7 +175,7 @@ export class DafnyInstaller {
   }
 
   private async installFromSource() {
-    const installationPath = this.getCustomInstallationPath(os.arch());
+    const installationPath = await this.getCustomInstallationPath(os.arch());
```

Comment:
There can now be a race between the custom installation path and the installation path (see below), because they both access a remote API at two different moments.
Could you please refactor the code as needed so that this API is invoked only once in this function?

Also, below at line 214, you have another call to
```
await this.execLog(`git checkout v${getConfiguredVersion()}`);
```
that you normally need to put an "await" in. But as per my comment above, if you ensure only one API call, you should be able to simplify it.
I'm looking forward the time when we will have tests to ensure such refactorings are not breaking the installation process.